tests/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1716,6 +1718,96 @@ private function createReservedKeywordTables(): void
         $schemaManager->createSchemaObjects($schema);
     }
 
+    /** @throws Exception */
+    public function testQuotedIdentifiers(): void
+    {
+        $platform = $this->connection->getDatabasePlatform();
+
+        if ($platform instanceof DB2Platform) {
+            self::markTestIncomplete(
+                'Introspection of lower-case identifiers as quoted is currently not implemented on IBM DB2.',
+            );
+        }
+
+        if (! $platform instanceof OraclePlatform && ! $platform instanceof PostgreSQLPlatform) {
+            self::markTestSkipped('The current platform does not auto-quote introspected identifiers.');
```

Comment:
This isn't a problem per se. Whether the given schema manager will quote certain identifiers or not, we want to make sure that they are introspected in the proper case. Unfortunately, there's no public API to get the original case of an index or constraint column. For example:
1. `Index::getColumns()` returns lower-cased column names.
2. `Index::getQuotedColumns()` quotes the column names only if they are explicitly marked as quoted (e.g. by Oracle or now PostgreSQL).

In DBAL 5, we should be able to drop this condition and run the test against all platforms consistently.